### DISCUSSION TITLE
Will the incorporation of WinUI 3 into File Explorer affect the shell object model?

### DISCUSSION THREAD

**Discussion by amtopel** [microsoft/microsoft-ui-xaml#8233] [Q&A]
Does anyone know if the upcoming new version of File Explorer, which will incorporate WinUI 3, will break the shell object model?  Will there still be `HWND`s for the shell views and whatnot?  And what about the UIAutomation tree/events?  Will those still be accessible like before?  Thanks for any info.

**Comment by DarranRowe** [comment]
Any publicly documented interfaces must always keep working. This is part of what supported means.
Anything internal is subject to change.
